flash/core/data/new_data_module.py

```diff
@@ -290,27 +295,20 @@ def _predict_dataloader(self) -> DataLoader:
             persistent_workers=self.persistent_workers,
         )
 
-    @property
-    def data_pipeline_state(self) -> DataPipelineState:
-        """Collect the states from the input datasets and their transforms."""
+    def connect(self, task: "flash.Task"):
         data_pipeline_state = DataPipelineState()
-        if self._train_input:
-            self._add_to_data_pipeline_state(self._train_input, data_pipeline_state)
-        if self._val_input:
-            self._add_to_data_pipeline_state(self._val_input, data_pipeline_state)
-        if self._test_input:
-            self._add_to_data_pipeline_state(self._test_input, data_pipeline_state)
-        if self._predict_input:
-            self._add_to_data_pipeline_state(self._predict_input, data_pipeline_state)
-        return data_pipeline_state
-
-    def _add_to_data_pipeline_state(self, input: Input, data_pipeline_state: DataPipelineState) -> None:
-        """Collect the states contained within the input datasets and their transforms."""
-        for state in input._state.values():
-            data_pipeline_state.set_state(state)
-        if isinstance(input.transform, InputTransform):
-            for state in input.transform._state.values():
-                data_pipeline_state.set_state(state)
+        for properties in [
+            self._train_input,
+            self._val_input,
+            self._test_input,
+            self._predict_input,
```

Comment:
Let's also add the transforms there or at least ensure broadcasting for the input to its transform.

Reply:
Done :smiley: